runtime/src/main/java/org/corfudb/runtime/LiteRoutingQueueListener.java

```diff
@@ -72,105 +77,95 @@ public LiteRoutingQueueListener(CorfuStore corfuStore, String sourceSiteId, Stri
+                .findFirst().get();
         List<Table.CorfuQueueRecord> allEntries = new ArrayList<>(entries.size());
-        Queue.ReplicationType currentType = Queue.ReplicationType.LAST_FULL_SYNC_ENTRY;
+
         for (CorfuStreamEntry entry : entries) {
-            if (!entry.getOperation().equals(CorfuStreamEntry.OperationType.UPDATE)) {
-                continue;
-            }
+            // The Source always 'adds' replicated data to the queue.  So the op type must be UPDATE
+            Preconditions.checkState(entry.getOperation() == CorfuStreamEntry.OperationType.UPDATE);
             Queue.CorfuGuidMsg key = (Queue.CorfuGuidMsg) entry.getKey();
             Queue.RoutingTableEntryMsg msg = (Queue.RoutingTableEntryMsg) entry.getPayload();
             Queue.CorfuQueueMetadataMsg metadataMsg = (Queue.CorfuQueueMetadataMsg) entry.getMetadata();
             allEntries.add(new Table.CorfuQueueRecord(key, metadataMsg, msg));
-            if (currentType.equals(Queue.ReplicationType.LAST_FULL_SYNC_ENTRY)) {
-                currentType = msg.getReplicationType();
-            } else if (!currentType.equals(msg.getReplicationType())) {
-                throw new LogReplicationClientException("Not expecting mixed event types "+ currentType +" vs "
-                    +msg.getReplicationType());
-            }
+            currentReplicationType = msg.getReplicationType();
         }
         if (allEntries.isEmpty()) {
             return;
         }
-        boolean entriesProcessed;
+        boolean entriesProcessed = false;
         long now = System.currentTimeMillis();
-        long whenQrecordWasCreated = CorfuGuid.getTimestampFromGuid(allEntries.get(0).getRecordId().getInstanceId(), now);
-        if (currentType.equals(Queue.ReplicationType.LOG_ENTRY_SYNC)) {
-            log.info("LRQRecvListener delivering {} LOG_ENTRY updates took {}ms end to end",
-                    allEntries.size(), now - whenQrecordWasCreated);
-            entriesProcessed = processUpdatesInLogEntrySync(allEntries.stream().sorted().map(q ->
-                    (Queue.RoutingTableEntryMsg)q.getEntry()).collect(Collectors.toList()));
-        } else {
-            log.info("LRQRecvListener delivering {} FULLSYNC updates took {}ms end to end",
-                    allEntries.size(), now - whenQrecordWasCreated);
-            entriesProcessed = processUpdatesInSnapshotSync(allEntries.stream().sorted().map(q ->
+        long whenQRecordWasCreated = CorfuGuid.getTimestampFromGuid(allEntries.get(0).getRecordId().getInstanceId(),
+            now);
+
+        switch (currentReplicationType) {
+            case LOG_ENTRY_SYNC:
+                log.info("LRQRecvListener delivering {} LOG_ENTRY updates took {}ms end to end",
+                    allEntries.size(), now - whenQRecordWasCreated);
+                entriesProcessed = processUpdatesInLogEntrySync(allEntries.stream().sorted().map(q ->
```

Comment:
Why would we need to sort the data here?

Reply:
yes, not sure why it was added